framework/include/geomsearch/PenetrationLocator.h

```diff
@@ -83,22 +81,36 @@ class PenetrationLocator : Restartable
   void setCheckWhetherReasonable(bool state);
   void setUpdate(bool update);
   void setTangentialTolerance(Real tangential_tolerance);
-  void setNormalSmoothingDistance(Real normal_smoothing_distance);
-  void setNormalSmoothingMethod(std::string nsmString);
+  /**
+   * Enable normal smoothing based on edge normals
+   * @param normal_smoothing_distance Distance from edge in parametric coordinates over which to
+   * smooth the normal
+   */
+  void setEdgeBaseSmoothingMethod(Real normal_smoothing_distance);
+  /**
+   * Enable normal smoothing based on nodal normals
+   * @param uo_name The name of the user object that provides the nodal normals
+   */
+  void setNodalNormalSmoothingMethod(const UserObjectName & uo_name);
   Real getTangentialTolerance() { return _tangential_tolerance; }
 
 protected:
   /// Check whether found candidates are reasonable
   bool _check_whether_reasonable;
-  bool & _update_location;         // Update the penetration location for nodes found last time
-  Real _tangential_tolerance;      // Tangential distance a node can be from a face and still be in
-                                   // contact
-  bool _do_normal_smoothing;       // Should we do contact normal smoothing?
-  Real _normal_smoothing_distance; // Distance from edge (in parametric coords) within which to
-                                   // perform normal smoothing
-  NORMAL_SMOOTHING_METHOD _normal_smoothing_method;
-
-  const Moose::PatchUpdateType _patch_update_strategy; // Contact patch update strategy
+  /// Update the penetration location for nodes found last time
+  bool & _update_location;
+  /// Tangential distance a node can be from a face and still be in contact
+  Real _tangential_tolerance;
+  /// Should we do contact normal smoothing?
+  bool _do_normal_smoothing;
```

Comment:
Notice I only fixed the style of comments, but I can rearrange the member variables

Reply:
So, why are the interweaved bools a problem in `PenetrationLocator`? We do not have a lot of them usually. I could see this would be a problem in `PenetrationInfo`...

Reply:
Not a problem, just a note.